Cookbook/Proposals/AdditionalRulesForDanger.md

```diff
@@ -0,0 +1,133 @@
+
+The algorithm is then be executed once more but for the `tearDown()` method instead.
+If the number of changes inside each function matches, then no issue would be found. If not, then Current appeared to have been mutated but not restored; we'd issue a warning for the lines in which the algorithm detected a change and alert the developers in GitHub.
+
+Nevertheless, this rule is rather naïve because we have no actual way of detecting if Current was reset or not; that would require compiling and running the actual Swift code. Therefore, this approach is not a silver bullet or anything remotely similar but it would still be useful for one use case in particular: should the developer forget to restore Current, this rule will alert us of this situation.
+
+### Translation errors
+
+Translations are particularly hard to catch during PR code reviews, especially if a lot of new phrases are added at once.
+While, in theory, we have a fastlane set up to deal with these translations there is always the risk that developers mistype the localization key when setting up a new static var.
+
+As such, I'd like to suggest another rule for Danger that would:
+    1. detect new static String variables in every Localizable.swift file.
+    2. if we detected a new static variable we would then scan the localization string file (`Babylon/Supporting Files/Localization/en.lproj/Localizable.strings`) and look for that key
+
+We would need to whitelist `Localizable.*\.swift` in order to make this work.
+
+### Feedbacks must have the `when` prefix
+
+Feedbacks should use the `when` prefix in their names.
+In order to detect violations of this practice, we could use the following regex `static func (?!when).* -> Feedback<State,`.
+Regarding the whitelist, we'd apply this rule to every ViewModel.
+
+Currently, our codebase contains 38 occurences spread across 29 files which makes it a good candidate for SwiftLint. 
+
+In any case, this rule's usefulness is limited once we consider the fact that our feedback functions are usually spread across multiple lines. The regex's complexity would increase a lot and, to be entirely honest, I'm unsure the benefits outweight the downsides.
+
+## Impact on existing codebase
+
+While our codebase itself would not suffer any impact at all, our code review process would now have additional comments that would warn us of a particular situation. Should a PR contain numerous anti-patterns that break our rules the bot could potentially spam the PR with these warnings, depending on the way Danger's rules are implemented.
+
+Furthermore, we would be increasing Danger's processing time - although for the time being this is nearly negligible (roughly 4 seconds) - so this is a minor drawback that we could easily live with, in my honest opinion.
+
+## Alternatives considered
+
+* Implement a subset of those rules (and determine the most adequate tool to detect infractions for each rule).
+
+* Don't implement these rules at all
```

Comment:
* Fix some of the existing warnings (maybe some of them are in great number but relatively simple to fix with search/replace for example?) first, so that we could then re-consider make new warnings like SwiftLint's be more visible and thus more useful.

* Use SwiftLint for some of the rules mentioned above, even if we already have violations for them in our current codebase, but keep them as warnings. Would increase the number of warnings again, but would also allow us to see warnings in any new code we write immediately, instead of only at PR time. Would also allow us to see warnings on existing code when we work on an existing file and hopefully take that occasion to refactor and fix the warning while working on our feature (e.g. if I work on an existing, legacy screen, if I only get the warning via Danger I might not be aware that it's legacy and I should take the occasion to refactor it a bit while I'm at it... until I've done my ticket and open the PR)

* Use SwiftLint and make the violations be error, but use `// swiftlint:disable` on existing code already violating the rules, so that only the new code gets the warnings. (I'm not fan of this solution, since it would risk to legitimize the use of `swiftlint:disable` and end up defeating the purpose of SwiftLint in the long run, but had to mention it for the sake of argument)

Reply:
Fair points, will revisit this section.